### PR TITLE
More its

### PR DESCRIPTION
**Pull request by tempit**

### CODE REVIEW

File: test/lib/service.spec.ts [hunk 1]
```diff
@@ -269,110 +585,282 @@ describe("Service component test", function () {
...
+            const fileSystem = new MemoryFileSystem('', { content: { [topFileName]: topFileText, [jsFileName]: jsFileText, [tsFileName]: tsFileText } });
+
+            init(fileSystem, testCon.server);
+            const context = { includeDeclaration: true }
+            const textDocument = TextDocumentItem.create(toVscodePath('/' + topFileName), 'stylable', 0, fileSystem.loadTextFileSync(topFileName));
+
+            const mixinPos = { line: 11, character: 29 };
+            const mixinRes = await testCon.client.signatureHelp({ textDocument, position: mixinPos });
+
+            const mixinSig: SignatureHelp = {
```
Comment (amir-arad) [inline test/lib/service.spec.ts]: This comment is for the rest of the signature tests:
there's a lot of boilerplate surrounding building expected signature response
extract common driver for that.
most of the code complexity deals with building  a single `SignatureInformation`, so maybe use a separate `expect` expression to match `response.signatures[0]`, as matching the rest of the response is a simple structure validation that can be extracted to a common function (activeSignature, activeParameter are almost always 0).

regarding simplifying the `SignatureInformation` matcher, notice how the signature is mostly composed of the params descriptions (repeated literal data), all parameters are string arguments mapped with `ParameterInformation.create`, the second argument is `undefined` and the entire thing is mapped with `SignatureInformation.create` -  write a builder that handles that outside of the test code.

File: test/lib/service.spec.ts [hunk 2]
```diff
@@ -269,110 +585,282 @@ describe("Service component test", function () {
...
+                signatures: [SignatureInformation.create(
+                    'Supported state types:\n- "string | number | enum | tag"',
+                    undefined,
+                    ParameterInformation.create("string | number | enum | tag")
+                )]
+            }
+            expect(stateParamTypesSigRes).to.eql(stateParamTypesSig);
+        }));
+
+        it('Signatures - State with params definition (validator)', plan(1, async () => {
```
Comment (amir-arad) [inline test/lib/service.spec.ts]: redundant prefix

File: test/lib/service.spec.ts [hunk 3]
```diff
@@ -41,8 +41,287 @@ describe("Service component test", function () {
...
+            |  -st-named: aFormatter;
+            |}
+            |
+            |.local {
+            |  color: aFormatter(red);
+            |}`
+
+            const importFileText = trimLiteral`
+            |/**
+            | * @description A mixin with no params
```
[the remark this answers is not in the document]
Comment (tempit) [reply test/lib/service.spec.ts]: The test is for formatter/mixin definitions.
A JS formatter/mixin definition in Stylable (as opposed to TS) has to have the docs for type signatures (e.g. so we know whether it's a mixin or a formatter). 
I'll rename the test to JS formatters (and add a TS test) , that should make it clear(er) to anyone who knows mixins/formatters in stylable.

File: test/lib/service.spec.ts [hunk 4]
```diff
@@ -121,54 +400,90 @@ describe("Service component test", function () {
...
-            color: color
-        }]);
-
-        expect(importDocColors).to.eql([{
-            range: range3,
-            color: color
-        }]);
-    }));
+    describe("Document Colors", function () {
+        it("Document Colors - local, vars, imported", plan(2, async () => {
```
[the remark this answers is not in the document]
Comment (tempit) [reply test/lib/service.spec.ts]: It's not, really. It depends on which reporter you are using, and when the errors are interspersed with the test names it makes it easier to identify which test you're on.

File: test/lib/service.spec.ts (hunk 1 above)
Comment (tempit) [reply test/lib/service.spec.ts]: I'm not sure I agree here. I think this format is more readable (though verbose) than a builder for a SignatureHelp object, which is more complex than the objects in other tests. We can talk about it more, leaving as is for now.

File: src/lib/provider.ts [hunk 5]
```diff
@@ -13,8 +13,9 @@ import {
     Stylable,
     StylableMeta,
     StylableTransformer,
-    valueMapping
+    valueMapping,
 } from 'stylable';
+import { nativeFunctionsDic } from 'stylable/dist/src/native-reserved-lists'
```
Comment (amir-arad) [inline src/lib/provider.ts]: import from index, not internals.
see @barak007 's [response](https://github.com/wix/stylable/issues/320#event-1648149468)

File: test/lib/service.spec.ts [hunk 6]
```diff
@@ -792,7 +778,7 @@ describe("Service component test", function () {
 
             const textDocument = TextDocumentItem.create(toVscodePath('/' + topFileName), 'stylable', 0, fileSystem.loadTextFileSync(topFileName));
             const position = { line: 1, character: 24 };
-            const stateParamTypesSigRes = await testCon.client.signatureHelp({ textDocument, position });
+            const res = await testCon.client.signatureHelp({ textDocument, position });
 
             const stateParamTypesSig: SignatureHelp = {
                 activeSignature: 0,
```
Comment (amir-arad) [inline test/lib/service.spec.ts]: delete unused

**Pull request merged**